mathesar/api/db/viewsets/records.py

```diff
@@ -103,7 +103,7 @@ def list(self, request, table_pk=None):
                 status_code=status.HTTP_400_BAD_REQUEST
             )
         except DataError as e:
-            if isinstance(e.orig, InvalidDatetimeFormat):
+            if isinstance(e.orig, (InvalidDatetimeFormat, DatetimeFieldOverflow)):
```

Comment:
Add an else to raise `e`, for the case in which we're unable to catch the specific exception.

Reply:
Do you mean that I should add an `except Exception as e` to handle all errors, or do you want me to create a separate `elif` statement for `DatetimeFieldOverflow`?

Reply:
Is this what you want?
![image](https://github.com/mathesar-foundation/mathesar/assets/34382211/070e8f8e-677d-4750-8db4-18c3c52fea69)

Reply:
Your current Implementation is fine just add an else block to throw any _other_ `DataError`, I want to eliminate the senarios where we are catching the exception correctly(`DataError` in this case) but not throwing it.

Reply:
Added these changes.